pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/MetadataFetcherFactory.java

```diff
@@ -28,7 +30,12 @@
  * abstract class which is returned by the MetadataFetcherFactory. 
  */
 public class MetadataFetcherFactory {
-    static public MetadataFetcher create(String fetcherName) throws Exception {
-        return (MetadataFetcher) Class.forName(fetcherName).newInstance();
+    /* TODO: This is a tempororary workaround.
+     * The metadata class will be moved to the pxf-profile.xml in the future
+     */
```

Comment:
btw, why is it temporary? currently, both profile and specific plugin names are supported - so it will probably be possible to define a profile that includes the metadata fetcher, or specify the metadata fetcher itself.